docs/advanced/pool.md

```diff
@@ -0,0 +1,90 @@
+# Custom Pool
+
+::: warning
+This is advanced API. If you are just running tests, you probably don't need this. It is primarily used by library authors.
+:::
+
+Vitest runs tests in pools. By default, there are several pools:
+
+- `threads` to run tests using `node:worker_threads` (isolation is provided with a new worker context)
+- `forks` to run tests using `node:child_process` (isolation is provided with a new `child_process.fork` process)
+- `vmThreads` to run tests using `node:worker_threads` (but isolation is provided with `vm` module instead of a new worker context)
+- `browser` to run tests using browser providers
+- `typescript` to run typechecking on tests
+
+You can provide your own pool by specifying a file path:
+
+```ts
+export default defineConfig({
+  test: {
+    // will run every file with a custom pool by default
+    pool: './my-custom-pool.ts',
+    // you can provide options using `poolOptions` object
```

Comment:
While testing this I tried to use the `test.pool` here in `test.poolOptions.<key>` as key. Might be good to mention that the key has to be pool's `name` property:

```suggestion
    // you can provide options using `poolOptions` object and name of the custom pool
```

Reply:
You can use any key, actually.

Reply:
Oh right, it just depends on how the pool itself is accessing the options.